core/utils/parsing.ts

```diff
@@ -132,11 +141,11 @@ function tokenizeInterpolationInternal(
     tokens.push(text);
   }
 
-  // Merge adjacent text tokens into a single string.
+  // Merge adjacent text tokens into a single string (excluding newline tokens).
```

Comment:
In the case that we're just replacing message references, the `tokens` array is of length `1` correct? So this still properly returns a `mergedTokens` array of length `1` in that case?

Just want to make sure I understand what this is doing :P

Reply:
I believe message references involving % characters will end up as separate elements in the tokens array above here, but this loop will those tokens via the buffer variable here, which will then become the only output token.